mindone/utils/load_params.py

```diff
@@ -0,0 +1,136 @@
+        if not isinstance(key, str) or not isinstance(value, (Parameter, str)):
+            logger.critical("Load parameters into net failed.")
+            msg = (
+                "For 'parameter_dict', the element in the argument 'parameter_dict' should be a "
+                "'str' and 'Parameter' , but got {} and {}.".format(type(key), type(value))
+            )
+            raise TypeError(msg)
+
+    # TODO: replace by otherway to do check_bool
+    # strict_load = Validator.check_bool(strict_load)
+    logger.info("Execute the process of loading parameters into net.")
+    net.init_parameters_data()
+    param_not_load = []
+    ckpt_not_load = list(parameter_dict.keys())
+    for _, param in net.parameters_and_names():
+        if param.name in parameter_dict:
+            new_param = copy.deepcopy(parameter_dict[param.name])
+            _update_param(param, new_param, strict_load)
+            ckpt_not_load.remove(param.name)
+        else:
+            param_not_load.append(param.name)
+
+    if param_not_load and not strict_load:
+        _load_dismatch_prefix_params(net, parameter_dict, param_not_load, strict_load)
+
+    logger.info("Loading parameters into net is finished.")
+    if filter:
+        param_all_load_flag = len(set(param_not_load).intersection(set(filter))) == 0
+        if param_all_load_flag:
+            param_not_load.clear()
+    if param_not_load:
+        logger.warning(
+            "For 'load_param_into_net', "
+            "{} parameters in the 'net' are not loaded, because they are not in the "
+            "'parameter_dict', please check whether the network structure is consistent "
+            "when training and loading checkpoint.".format(len(param_not_load))
+        )
+        for param_name in param_not_load:
+            logger.warning("{} is not loaded.".format(param_name))
+    return param_not_load, ckpt_not_load
+
+
+def load_checkpoint_to_net(
+    net: nn.Cell,
+    checkpoint: Union[str, dict],
+    ignore_net_params_not_loaded=False,
+    ensure_all_ckpt_params_loaded=False,
+):
+    """
+    ignore_net_params_not_loaded: set True for inference if only a part of network needs to be loaded, the flushing net-not-loaded warnings will disappear.
+    ensure_all_ckpt_params_loaded : set True for inference if you want to ensure no checkpoint param is missed in loading
+    """
+    if isinstance(checkpoint, str):
+        if os.path.exists(checkpoint):
+            param_dict = ms.load_checkpoint(checkpoint)
+        else:
+            raise FileNotFoundError(f"{checkpoint} doesn't exist")
+    elif isinstance(checkpoint, dict):
+        param_dict = checkpoint
+    else:
+        raise TypeError(f"unknown checkpoint type: {checkpoint}")
+
+    if param_dict:
+        if ignore_net_params_not_loaded:
+            filter = param_dict.keys()
+        else:
+            filter = None
+        param_not_load, ckpt_not_load = load_param_into_net_with_filter(net, param_dict, filter=filter)
```

Comment:
```suggestion
        if ignore_net_params_not_loaded:
            filter = [param.name for _, param in net.parameters_and_names() if param.name not in parameter_dict]
            for param in filter:
                parameter_dict.pop(param)
        param_not_load, ckpt_not_load = ms.load_param_into_net(net, param_dict)
```

Reply:
As mentioned above, it's for filtering network param to be loaded instead